src/main/java/seedu/address/model/person/Person.java

```diff
@@ -47,26 +49,53 @@ public Person(Name name, Phone phone, Email email, Address address, Set<Tag> tag
     public Person(Name name, Phone phone, Email email, Address address, Set<Tag> tags) {
         requireAllNonNull(name, phone, email, address, tags);
         this.name = name;
-        this.phone = phone;
-        this.email = email;
-        this.address = address;
+        this.phone = Optional.of(phone);
+        this.email = Optional.of(email);
+        this.address = Optional.of(address);
         this.tags.addAll(tags);
     }
 
+    /**
+     * Creates a code{Person} with the specified code{attribute} using the input code{person}.
+     * Other than name and tags, other unspecified field is empty.
+     */
+    public Person(Person person, Attribute attribute) {
+        requireAllNonNull(person, attribute);
+        this.name = person.name;
+        if (attribute.equals(Attribute.POLICY_ID)) {
+            this.phone = Optional.empty();
+            this.email = Optional.empty();
+            this.address = Optional.empty();
+            this.policies.addAll(person.policies);
+        } else if (attribute.equals(Attribute.PHONE)) {
+            this.phone = Optional.of(person.getPhone().get());
+            this.email = Optional.empty();
+            this.address = Optional.empty();
+        } else if (attribute.equals(Attribute.ADDRESS)) {
+            this.phone = Optional.empty();
+            this.email = Optional.empty();
+            this.address = Optional.of(person.getAddress().get());
+        } else {
+            this.phone = Optional.empty();
+            this.email = Optional.of(person.getEmail().get());
+            this.address = Optional.empty();
+        }
```

Comment:
Same here. Suggestion to use switch statements.

Reply:
Done!